### PR TITLE
Rails 5.1

### LINKED ISSUE #1143: Move to rails 5.1
Thanks for creating an issue! Please fill out this form so we can be sure to have all the information we need, and to minimize back and forth.

* **What are we trying to do?**

Upgrade to rails 5.1

* **What feature or behavior is this required for?**

Staying current! Long term health of the project! Using systemtest!!!

* **How could we solve this issue? (Not knowing is okay!)**

Put it in the capable hands of @DarthHater and let 'er rip.

* **Anything else?**

nope!

### PR DESCRIPTION
**Pull request by DarthHater**
I rule and have completed some work on Case Manager that's ready for review!

This brings us up to Rails 5.1

This pull request makes the following changes:
* Gemfile updates
* Gemfile.lock updates

It relates to the following issue #s: 
* Fixes #1143

### CODE REVIEW

File: bin/rails [hunk 1]
```diff
@@ -1,10 +1,4 @@
 #!/usr/bin/env ruby
-begin
-  load File.expand_path('../spring', __FILE__)
-rescue LoadError => e
-  raise unless e.message.include?('spring')
-end
-APP_PATH = File.expand_path('../../config/application', __FILE__)
-
+APP_PATH = File.expand_path('../config/application', __dir__)
```
Comment (colinxfleming) [inline bin/rails]: Didn't they just change this in 5.0? man...

File: circle.yml [hunk 2]
```diff
@@ -19,8 +19,7 @@ deployment:
 
 test:
   override:
-    - bundle exec rake knapsack:minitest:
-        parallel: true
+    - bundle exec rails test
```
Comment (colinxfleming) [inline circle.yml]: Just a note to us that we should restore parallelization before we declare this good to go. But definitely totally fine until we figure out that weird knapsack error.

File: config/environments/development.rb [hunk 3]
```diff
@@ -9,59 +9,43 @@
...
   # Suppress logger output for asset requests.
   config.assets.quiet = true
 
+  # Raises error for missing translations
+  # config.action_view.raise_on_missing_translations = true
+
   # Use an evented file watcher to asynchronously detect changes in source code,
   # routes, locales, etc. This feature depends on the listen gem.
-  # config.file_watcher = ActiveSupport::EventedFileUpdateChecker
+  config.file_watcher = ActiveSupport::EventedFileUpdateChecker
```
Comment (colinxfleming) [inline config/environments/development.rb]: kinda nice!

File: config/environments/development.rb [hunk 4]
```diff
@@ -9,59 +9,43 @@
   # Do not eager load code on boot.
```
Comment (colinxfleming) [inline config/environments/development.rb]: Looks like this is just everything slightly reorganized, except for `public_file_server_enabled` and `actionmailer.default_url_options` being removed.

File: config/initializers/assets.rb [hunk 5]
```diff
@@ -1,11 +1,14 @@
 # Be sure to restart your server when you modify this file.
 
 # Version of your assets, change this if you want to expire all your assets.
-Rails.application.config.assets.version = '2017-03-26'
+Rails.application.config.assets.version = '1.0'
```
Comment (colinxfleming) [inline config/initializers/assets.rb]: Probably want to swap this with a current date, but cool either way.

File: config/initializers/new_framework_defaults.rb [hunk 6]
```diff
@@ -20,4 +20,4 @@
 # Rails.application.config.active_record.belongs_to_required_by_default = false
 
 # Do not halt callback chains when a callback returns false. Previous versions had true.
-ActiveSupport.halt_callback_chains_on_return_false = true
+ActiveSupport.halt_callback_chains_on_return_false = false
```
Comment (colinxfleming) [inline config/initializers/new_framework_defaults.rb]: So I'm PRETTY sure this was a mongoid thing but I can't remember. Definitely something to check on but I'm pretty sure this is okay.

File: config/initializers/new_framework_defaults_5_1.rb [hunk 7]
```diff
@@ -0,0 +1,14 @@
+# Be sure to restart your server when you modify this file.
+#
+# This file contains migration options to ease your Rails 5.1 upgrade.
+#
+# Once upgraded flip defaults one by one to migrate to the new default.
+#
+# Read the Guide for Upgrading Ruby on Rails for more info on each option.
+
+# Make `form_with` generate non-remote forms.
+Rails.application.config.action_view.form_with_generates_remote_forms = false
```
Comment (colinxfleming) [inline config/initializers/new_framework_defaults_5_1.rb]: Interesting decision by the rails team here to move away from synchronous forms by default. We'll want to check with our `rails_bootstrap_forms` library to make sure it starts taking advantage of this change.

File: config/application.rb [hunk 8]
```diff
@@ -15,6 +15,8 @@
 
 module DcafCaseManagement
   class Application < Rails::Application
+    # Initialize configuration defaults for originally generated Rails version.
+    config.load_defaults 5.1
```
Comment (colinxfleming) [inline config/application.rb]: I'm a LITTLE skeptical of this but let's roll with it!

Comment (DarthHater) [inline config/application.rb]: Yeah it was a new value from rails app:update. I think it just tells it to load new_framework_defaults and then the version.

File: config/environments/development.rb (hunk 3 above)
Comment (DarthHater) [inline config/environments/development.rb]: Yeah, glorified grunt watch or whatever but still gonna be cool I think

File: bin/rails (hunk 1 above)
Comment (DarthHater) [inline bin/rails]: I think so, which I kinda laughed at too

**Comment by DarthHater** [comment]
@colinxfleming I'll tackle the tests if you want to take a look at knapsack.

**Comment by colinxfleming** [comment]
on it!

**Comment by colinxfleming** [comment]
Knapsack / CI gem in #1177

**Comment by colinxfleming** [comment]
We're getting flaky test failures at this point, which is good, I guess. @DarthHater for review

**Comment by colinxfleming** [comment]
@DarthHater if this looks good to you, let's SMASH THAT MERGE BUTTON and work on the other components in separate PRs per the plan.

**Comment by colinxfleming** [comment]
eat it build.

**Comment by DarthHater** [comment]
I'll take a gander at this tomorrow. Thanks for pushing it forward while I was being the road warrior.

**Comment by colinxfleming** [comment]
Swish. Yeah, if this passes muster we should move as is and then port over the tests incrementally in their own PRs.

**Comment by colinxfleming** [comment]
@DarthHater ping for a review on this if you have a minute. I'd like to get some of this moving as I think the move to systemtest for some of our problem children will fix some of our flaky test garbage.

File: config/initializers/new_framework_defaults_5_1.rb (hunk 7 above)
Comment (colinxfleming) [reply config/initializers/new_framework_defaults_5_1.rb]: https://github.com/bootstrap-ruby/rails-bootstrap-forms/pull/346 is the relevant PR but we won't have to worry about implementing for awhile.

**Comment by colinxfleming** [comment]
Build is passing. Some local poking at comes back clean. I'm declaring victory and merging so we can get this into sandbox.

**Pull request merged**